test/kt/CMakeLists.txt

```diff
@@ -0,0 +1,105 @@
+    endif()
+
+    add_custom_target(run-${_target_name}
+        COMMAND "${CMAKE_CTEST_COMMAND}" -R ^${_target_name}$$ --output-on-failure --no-label-summary
+        USES_TERMINAL
+        DEPENDS ${_target_name}
+        COMMENT "Build and run test ${_target_name}")
+endfunction(_generate_test)
+
+function(_generate_test_randomly _target_name _test_path _probability_permille)
+    string(RANDOM LENGTH 3 ALPHABET "0123456789" _random)
+    if ((${_random} LESS ${_probability_permille}) AND (NOT TARGET ${_target_name}))
+        _generate_test(${_target_name} ${_test_path})
+    endif()
+endfunction()
+
+# Arguments with special treatment:
+#   _val_type. Treated as absent for values: "" (empty string).
+#              If absent, key sort test is generated, otherwise key-value sort test
+function(_generate_esimd_sort_test _data_per_work_item _work_group_size _key_type _val_type _probability_permille)
+    if ((NOT TARGET build-esimd-tests) AND (NOT TARGET run-esimd-tests))
+        add_custom_target(build-esimd-tests COMMENT "Build all ESIMD-based kernel template tests")
+        add_custom_target(run-esimd-tests
+            COMMAND "${CMAKE_CTEST_COMMAND}" -R "^run-esimd-tests$" --output-on-failure --no-label-summary
+            DEPENDS build-esimd-tests
+            COMMENT "Build and run all ESIMD tests")
+    endif()
+
+    string(REPLACE "_t" "" _key_type_short ${_key_type})
+    if (NOT "${_val_type}" STREQUAL "")
+        string(REPLACE "_t" "" _val_type_short ${_val_type})
+        set(_target_name "esimd_radix_sort_dpwi${_data_per_work_item}_wgs${_work_group_size}_${_key_type_short}_${_val_type_short}")
+        set(_test_path "esimd_radix_sort_by_key.cpp")
+    else()
+        set(_target_name "esimd_radix_sort_dpwi${_data_per_work_item}_wgs${_work_group_size}_${_key_type_short}")
+        set(_test_path "esimd_radix_sort.cpp")
+    endif()
+
+    _generate_test_randomly(${_target_name} ${_test_path} ${_probability_permille})
+    if(TARGET ${_target_name})
+        add_dependencies(build-esimd-tests ${_target_name})
+        add_dependencies(run-esimd-tests ${_target_name})
+
+        target_compile_definitions(${_target_name} PRIVATE TEST_DATA_PER_WORK_ITEM=${_data_per_work_item})
+        target_compile_definitions(${_target_name} PRIVATE TEST_WORK_GROUP_SIZE=${_work_group_size})
+        target_compile_definitions(${_target_name} PRIVATE TEST_KEY_TYPE=${_key_type})
+        if (NOT "${_val_type}" STREQUAL "")
+            target_compile_definitions(${_target_name} PRIVATE TEST_VALUE_TYPE=${_val_type})
+        endif()
+    endif()
+endfunction()
+
+function(_generate_esimd_sort_tests _key_value_pairs)
+    set(_data_per_work_item_all "32" "64" "96" "128" "160" "192" "224" "256" "288" "320" "352" "384" "416" "448" "480" "512")
+    set(_work_group_size_all "32" "64")
+    set(_type_all "char" "uint16_t" "int" "uint64_t" "float" "double")
```

Comment:
I think its worth adding some comments here to explain / justify the strategy here for generating / separating test cases, and randomly selecting a subset.

Eventually, it may be useful in other places, and we may want to abstract it out to reuse, but not in this PR.